frame/uniques/src/functions.rs

```diff
@@ -336,7 +336,7 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 				Collection::<T, I>::get(&collection_id).ok_or(Error::<T, I>::UnknownCollection)?;
 			let transfer_amount = royalties * amount;
 
-			amount_left = amount - transfer_amount;
+			amount_left = amount.saturating_sub(transfer_amount);
```

Comment:
Note that i didn't comment on this because it is impossible for `transfer_amount` to be more than `amount` since `transfer_amount` is the result of `royalties * amount`, and `royalties` is always 1 or less... but anyway this is a fine change imo